### ISSUE TITLE
A bug when running runWarpLDAModels() with exactly one topic

### ISSUE THREAD

**Issue by Van1yu3** [aertslab/cisTopic#47]
Hi!

I would like to report a bug when running `runWarpLDAModels()` with exactly one topic.

The error message is 

> Error in alpha/t : non-numeric argument to binary operator

After checking the R code, I finally located the error, which is in [line 644 of RunModels.R](https://github.com/aertslab/cisTopic/blob/70291db487f35d65a7d29ce3a03de7498cbdb34b/R/RunModels.R#L644). The `t` in this line should be `topic`, I think.

Best
Van1yu3

**Comment by Van1yu3** [comment]
Also, the `t` should be `topic` in [line 638](https://github.com/aertslab/cisTopic/blob/70291db487f35d65a7d29ce3a03de7498cbdb34b/R/RunModels.R#L638).

**Comment by cbravo93** [comment]
Hi @Van1yu3 !

Thanks a lot for noticing, it is corrected now! The bug was affecting 644 and 647.

Cheers,

C

**Comment by Van1yu3** [comment]
Hi @cbravo93 !

Thank you for your reply. Sadly I still got some problems when the topic is exactly one.

Indeed, there are two issues when I run `runWarpLDAModels()` with exacly one topic:

1.  I got the following error message:
> Error: $ operator is invalid for atomic vectors
I guess this error is from [line 651](https://github.com/aertslab/cisTopic/blob/99a1173dcdbf74f1387714c9dd6527dff0ab39ee/R/RunModels.R#L651) , where the `y` is not what `topic_sums` is expected to be taken from.

2. The random seed seemed not to function since I found I got 2 different results with the same random seed. This happens probably because you forgot to `set.seed()` in the `runWarpLDAModels()` somewhere. I only see `set.seed()` function in [line 160](https://github.com/aertslab/cisTopic/blob/99a1173dcdbf74f1387714c9dd6527dff0ab39ee/R/RunModels.R#L160) in the `runCGSModels()`.
